kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -250,6 +252,18 @@ public void uploadPart(ByteArrayInputStream inputStream, int partSize) {
                                             .withPartNumber(currentPartNumber)
                                             .withPartSize(partSize)
                                             .withGeneralProgressListener(progressListener);
+
+      if (connectorConfig.uploadIntegrityCheck()) {
+        try {
+          String digest = Base64.getEncoder().encodeToString(DigestUtils.md5(inputStream));
+          // Reset the marker so that the stream may be read again while uploading
+          inputStream.reset();
+          request.setMd5Digest(digest);
```

Comment:
This message gets rejected by AWS. 

This approach worked for me:
- Create a `MessageDigest` https://github.com/nextflow-io/nextflow/blob/d8f8bf56cfe79dc05db7c149dafa0174717ae7f4/plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3OutputStream.java#L227
- Update it any time we put bytes in the buffer https://github.com/nextflow-io/nextflow/blob/d8f8bf56cfe79dc05db7c149dafa0174717ae7f4/plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3OutputStream.java#L266
- Send the digest encoded to string with the payload https://github.com/nextflow-io/nextflow/blob/d8f8bf56cfe79dc05db7c149dafa0174717ae7f4/plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3OutputStream.java#L329